text/0039-service-overhaul.md

```diff
@@ -0,0 +1,462 @@
+- "More Info" button
+
+## Service plugins
+### Issue
+#### `rtmp-services`
+This plugin provide services with only one protocol, so it needs to "register" service like YouTube multiple times (HLS & RTMPS).
+
+And also OBS is too heavily dependent of it.
+
+#### Service integrations
+Those integrations are added and use `rtmp-services` in a hacky way and not as plugins.
+
+### Idea for new services plugins
+#### `rtmp-services`
+This plugin will be put in a state of deprecation. And be replaced and finally removed after one major release cycle.
+
+Since OBS is heavily dependent of this plugin, many changes will be needed to remove this dependency.
+
+##### Conversion `rtmp_common` and `rtmp_custom` to new services
+A way to convert user service config to a new one will be added. More detail below.
+
+#### Service integrations
+
+Twitch, YouTube and Restream have their own integration in OBS with OAuth. But those need to be isolated as plugins (frontend if needed).
+
+Those will plugins will provide the basic and OAuth version of the services as one services.
+
+If OBS is built without client ids and hashes, those plugins will be built without their integration.
+
+##### Conversion to new plugins
+Everything related to auth in `basic.ini` will moved elsewhere. More detail below.
+
+##### About dock additions (idea/draft)
+A new function in the Service API to load UI additions when starting OBS or applying setting. An unload may be needed or added to the destroy function.
+
+##### About "Manage Broadcast" in YouTube integration About dock additions (idea/draft)
+YouTube adds a "broadcast manager", some services could also have the same need (e.g. PeerTube) so the button will be kept and a new function and a flag to the service API allowing this behavior will be added.
+
+#### `custom-service`
+This plugin is meant to provide a replacement for `rtmp_custom` type.
+
+If OBS need to be heavily dependent to one plugin, it shall be this one.
+
+The protocol will be detected (or maybe forced set by the user).
+Maybe properties will show depending of the protocol.
+
+#### `obs-services`
+This plugin is meant to provide a replacements for `rtmp_common` type for services who doesn't have custom behavior or integration.
+
+This plugin will need a brand new services.json with new format, to register each streaming service with their own id. No more things like `rtmp-common` id.
+ 
+Those services will be able to provide multiple protocols so no more "Service - HLS" and "Service - FTL".
+
+If a certain protocol is not available, the plugin will not register the service if the service doesn't use another protocol.
+If it does, the missing protocol will not be shown. [Thanks to RFC 45](https://github.com/tytan652/rfcs/blob/protocol_api/text/0045-protocols-in-outputs-api.md).
+
+The notion of maximum and recommended shall be distinguished.
+Maximum, supported resolutions, recommended settings will be ignorable with their respective checkboxes.
```

Comment:
The fewer checkboxes the better. Especially if the idea is to also have a properties view.

Reply:
- How about maximum are limits that even a user can't bypass and recommendation are ignore-able.
- Or recommendation are used to set settings with the Auto Wizard and maximums are ignore-able.

Reply:
This separation was removed, it still now a unique checkbox in the OBS Studio UI.